api/v1beta1/specialresource_types.go

```diff
@@ -137,12 +137,27 @@ type SpecialResourceDependency struct {
 	Set unstructured.Unstructured `json:"set,omitempty"`
 }
 
+// These are valid conditions of a SpecialResource.
+const (
+	// Ready means the SpecialResource is operational, ie. the recipe was reconciled.
+	SpecialResourceReady string = "Ready"
+
+	// Progressing means handling of the SpecialResource is in progress.
+	SpecialResourceProgressing string = "Progressing"
+
+	// Errored means SpecialResourceOperator detected an error that might be short-lived or unrecoverable without user's intervention.
+	SpecialResourceErrored string = "Errored"
+)
+
 // SpecialResourceStatus is the most recently observed status of the SpecialResource.
 // It is populated by the system and is read-only.
 // More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
 type SpecialResourceStatus struct {
-	// State describes at which step the chart installation is.
-	State string `json:"state"`
+	// Conditions contain observations about SpecialResource's current state
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
```

Comment:
No worries.
Since this is part of `.Status` and not `.Spec` I don't think we'd break anything. `.Status` is not something user should create when applying CR. 
Unless there's some other Operator programmatically watching CR's `.Status` before doing its own thing? I'd argue that that the `.Status.State` wasn't particularly good object to observe ;)

We can leave it (it sounds sensibly and responsibly), but on the other hand now is the best time to make (potentially) breaking changes (the later in upstreaming process the harder it'll be).
How about removing the code (leave it empty at all times) or leaving the code (then what will be the best moment to deprecate and remove this)?

Reply:
Brought `State` back to life

Reply:
Thank you!